schema/json.go

```diff
@@ -63,13 +64,24 @@ func (dst *JSON) Set(src interface{}) error {
 		} else {
 			*dst = JSON{Bytes: value, Status: Present}
 		}
+
 	// Encode* methods are defined on *JSON. If JSON is passed directly then the
 	// struct itself would be encoded instead of Bytes. This is clearly a footgun
 	// so detect and return an error. See https://github.com/jackc/pgx/issues/350.
 	case JSON:
 		return errors.New("use pointer to JSON instead of value")
 
 	default:
+		if isEmptyStringMap(value) {
+			*dst = JSON{Bytes: []byte("{}"), Status: Present}
+			break
+		}
+
+		if isEmptySlice(value) {
+			*dst = JSON{Bytes: []byte("[]"), Status: Present}
+			break
+		}
+
 		buf, err := json.Marshal(value)
```

Comment:
Let's do a small improvement in order:

1: do the `marshal` first. 
2: if error return error
3: check if the `buff` is null
4) if null do this check and turn `null` into `{}` or `[]`